risc0/zkvm/sdk/rust/src/receipt.rs

```diff
@@ -88,6 +99,13 @@ impl Receipt {
         self.verify_with_hal(&hal, method_id)
     }
 
+    /// This function is called by [verify](Receipt::verify), which provides the
+    /// CPU HAL generated using the program circuit.
```

Comment:
IIUC,  this is the generalized verify, which allows the caller to select whatever HAL they like (whereas `verify` uses the default SHA & CPU HAL). The way this comment is framed feels a bit backwards to me -- it doesn't exist to be called by `verify`, `verify` exists to provide default argument to `verify_with_hal`.